src/java/org/apache/cassandra/index/sai/disk/v1/segment/SegmentBuilder.java

```diff
@@ -73,103 +67,53 @@ public abstract class SegmentBuilder
     private ByteBuffer minTerm;
     private ByteBuffer maxTerm;
 
-    final IndexTermType indexTermType;
+    final StorageAttachedIndex index;
     long totalBytesAllocated;
     int rowCount = 0;
     int maxSegmentRowId = -1;
 
-    public static class BlockBalancedTreeSegmentBuilder extends SegmentBuilder
+    public static class RamBufferSegmentBuilder extends SegmentBuilder
     {
-        private final byte[] scratch;
-        private final BlockBalancedTreeRamBuffer trieBuffer;
+        protected final SegmentTrieBuffer segmentRamBuffer;
```

Comment:
The var should be named `segmentTrieBuffer`, in line with the new class name. Or maybe just `buffer`?